cluster/examples/kubernetes/ceph/common.yaml

```diff
@@ -76,12 +79,124 @@ spec:
+                          journalSizeMB:
+                            type: string
+                          osdsPerDevice:
+                            type: string
+                          encryptedDevice:
+                            type: string
+                            pattern: ^(true|false)$
+                      useAllDevices: 
+                        type: boolean
+                      deviceFilter:
+                        type: string
+                      devices:
+                        type: array
+                        items:
+                          properties:
+                            name:
+                              type: string
+                            config:
+                              properties:
+                                metadataDevice:
+                                  type: string
+                                storeType:
+                                  type: string
+                                  pattern: ^(filestore|bluestore)$
+                                databaseSizeMB:
+                                  type: string
+                                walSizeMB:
+                                  type: string
+                                journalSizeMB:
+                                  type: string
+                                osdsPerDevice:
+                                  type: string
+                                encryptedDevice:
+                                  type: string
+                                  pattern: ^(true|false)$
+                      location: 
+                        type: string
                   type: array
-                useAllDevices: {}
-                useAllNodes:
+                useAllDevices: 
                   type: boolean
+                deviceFilter:
+                  type: string
+                location: {}
+                config:
+                  properties:
+                    metadataDevice:
+                      type: string
+                    storeType:
+                      type: string
+                      pattern: ^(filestore|bluestore)$
+                    databaseSizeMB:
+                      type: string
+                    walSizeMB:
+                      type: string
+                    journalSizeMB:
+                      type: string
+                    osdsPerDevice:
+                      type: string
+                    encryptedDevice:
+                      type: string
+                      pattern: ^(true|false)$
+                topologyAware: 
+                  type: boolean
+            monitoring:
+              properties:
+                enabled:
+                  type: boolean
+                rulesNamespace:
+                  type: string
+            rbdMirroring:
+              properties:
+                workers:
+                  type: integer
+            placement:
+              properties:
+                all:
+                  properties:
```

Comment:
How does the validation help with the empty declaration?

Reply:
> empty declaration

Can you elaborate? Sorry for my ignorance.

Reply:
fixed

Reply:
I was referring to a declaration like this:
```
                placement: {}
```
Does that just mean it's an arbitrary struct, right?